lib/sign-request.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+# -What-------------------------------------------------------------------------
+# This script creates a request to the AWS Security Token Service API
+# with the action "GetCallerIdentity" and then signs the request using the
+# AWS credentials. It was modified from the python 2.x example published by
+# J. Thompson, the author of the Vault IAM auth method, at the vault support
+# mailing list. https://groups.google.com/forum/#!topic/vault-tool/Mfi3O-lW60I
+# -Why--------------------------------------------------------------------------
+# We are using python here instead of bash to take advantage of the boto3 library
+# which facilitates this work by an order of magnitude
+# -What-for---------------------------------------------------------------------
+# This is useful for authenticating to Vault, because a client can use
+# this script to generate this request and this request is sent with the
+# login attempt to the Vault server. Vault then executes this request and gets
+# the response from GetCallerIdentity, which tells who is trying to authenticate
+# ------------------------------------------------------------------------------
```

Comment:
Thank you for writing such an excellent explanatory comment! I'm going to try to copy this style for my own work.